### PR TITLE
docs: First sketch of OpenAPI definition for the key broker

### PR DESCRIPTION
**Pull request by paulhowardarm**
First cut of REST contract for the demo key broker.

### CODE REVIEW

File: doc/api/keys/keys.yaml [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+          type: string
+          description: Key Algorithm
+        n:
+          type: string
+          description: Key modulus
+        e:
+          type: string
+          description: Key exponent
+      description: >-
+        A JSON Web Key (https://www.rfc-editor.org/rfc/rfc7517) formatted RSA Public Key.
```
Comment (thomas-fossati) [inline doc/api/keys/keys.yaml:161]: Makes total sense.  ECIES is not (yet) well supported it seems, at least go is behind.

File: doc/api/keys/keys.yaml [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+    ErrorInformation:
+      required:
+        - type
+        - detail
+      properties:
+        type:
+          description: >-
+            https://github.com/confidential-containers/kbs/errors/<problem-type>
```
Comment (thomas-fossati) [inline doc/api/keys/keys.yaml]: Are these errors documented?  If not, it's better to create our error namespace.

File: doc/api/keys/keys.yaml [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+          type: string
+        detail:
+          type: string
+      description: >-
+        A Problem Details for HTTP APIs (https://www.rfc-editor.org/rfc/rfc7807)
```
Comment (thomas-fossati) [inline doc/api/keys/keys.yaml]: IETF nerd comment: this is https://www.rfc-editor.org/rfc/rfc9457 now :-)

File: doc/api/keys/keys.yaml [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+    ErrorInformation:
+      required:
+        - type
+        - detail
+      properties:
+        type:
+          description: >-
+            https://github.com/confidential-containers/kbs/errors/<problem-type>
+          type: string
```
Comment (thomas-fossati) [inline doc/api/keys/keys.yaml:169]: the type should be `uri`

Comment (thomas-fossati) [reply doc/api/keys/keys.yaml:169]: Besides, there are two options/schools:
1. self-documenting types: using URI with `https` scheme which can be de-referenced to a page explaining the error type;
2. non-self-documenting types: using URI with `tag` scheme.

See https://www.rfc-editor.org/rfc/rfc9457#section-3.1.1

**Review by thomas-fossati — APPROVED** [review]
Looks very good to me.

I have left a couple of suggestions in-line.

File: doc/api/keys/keys.yaml (hunk 2 above)
Comment (paulhowardarm) [reply doc/api/keys/keys.yaml]: That's a copy/paste mistake. I grabbed the simple type/detail tuple from the CoCo API, but accidentally left the description, which doesn't belong here. I'll delete that for now. Once we have enough information to be able to enumerate error type strings reliably, we can add those here.

File: doc/api/keys/keys.yaml (hunk 4 above)
Comment (paulhowardarm) [reply doc/api/keys/keys.yaml:169]: Swagger doesn't like "URI" as a type, but it looks like I can add a `format` directive while keeping the type as `string`.

**Pull request merged**